src/tests/StatusReporterTests.cpp

```diff
@@ -0,0 +1,35 @@
+#include "MockProducer.h"
+#include <StatusReporter.h>
+#include <gtest/gtest.h>
+#include <trompeloeil.hpp>
+
+using namespace std::chrono_literals;
+using trompeloeil::_;
+
+class StatusReporterTest : public ::testing::Test {};
+
+namespace Forwarder {
+TEST(StatusReporterTest, StatusReporterCallsProduce) {
+  std::chrono::milliseconds Interval(10);
+  MainOpt MainOptions;
+  std::string TopicName = "Topic Name";
+  KafkaW::BrokerSettings Settings;
+  auto KafkaProducer =
+      std::shared_ptr<KafkaW::Producer>(new KafkaW::MockProducer(Settings));
+  auto MockKafkaProducer =
+      dynamic_cast<KafkaW::MockProducer *>(KafkaProducer.get());
+  Streams streams;
+  std::unique_ptr<KafkaW::ProducerTopic> ApplicationStatusProducerTopic;
+  ApplicationStatusProducerTopic =
+      std::make_unique<KafkaW::ProducerTopic>(KafkaProducer, TopicName);
+
+  REQUIRE_CALL(*MockKafkaProducer, produce(_, _, _, _, _, _, _, _))
+      .TIMES(AT_LEAST(1))
+      .RETURN(RdKafka::ErrorCode::ERR_NO_ERROR);
+
+  StatusReporter TestStatusReporter(Interval, MainOptions,
+                                    ApplicationStatusProducerTopic, streams);
+  std::this_thread::sleep_for(100ms);
```

Comment:
The sleep can safely be reduced to `50ms` here and in the similar MetricsReporter test, and I think it is well worth those few milliseconds to catch regressions in these Reporter classes. We could keep track of how many times the timer has triggered and poll that in the test, but I question whether it us really worth Dolica's time and additional code to save a few milliseconds here when there are currently a load of bad tests which wait 5 seconds for metadata query timeout.

Reply:
I've created DM-1836 to look into stopping the tests trying to do network IO, which is costing 30 seconds of runtime.

Reply:
This is not about there being a 100ms wait. This is about the test failing because the wait is too short. At the same time, increasing the time to 500ms or 1s is too much I think and it might still fail the test in extreme cases.
There are ways of working around this non-determinism but it is not necessarily easy to do so. ASIO does make it possible though.

Reply:
The constructor makes the first call to the reporting function, so I think the runtime of the reporting function has to be comparable to the wait time for there to be risk of the test failing.

Reply:
That is not quite correct. The first call to the reporting function is currently not done in the constructor. Having the constructor do this is however a potential good solution to the problem. Instead of the `AsioTimer.async_wait()`call, call `reportStatus()` before starting the thread.